api/addon/v1alpha1/addon_types.go

```diff
@@ -381,3 +381,30 @@ func (a *Addon) CalculateChecksum() string {
 func (a *Addon) GetInstallStatus() ApplicationAssemblyPhase {
 	return a.Status.Lifecycle.Installed
 }
+
+func (p ApplicationAssemblyPhase) Completed() bool {
+	switch p {
+	case Succeeded, Failed:
+		return true
+	default:
+		return false
+	}
+}
+
+func (p ApplicationAssemblyPhase) Succeeded() bool {
+	switch p {
+	case Succeeded:
+		return true
+	default:
+		return false
+	}
```

Comment:
Couldn't this be `return p == Succeeded` ?

Reply:
The only difference is this way provides extensible. i.e if you add "Passed", do not need to change code